sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -598,6 +606,15 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
       IntervalUtils.castStringToYMInterval(s, it.startField, it.endField))
     case _: YearMonthIntervalType => buildCast[Int](_, s =>
       IntervalUtils.periodToMonths(IntervalUtils.monthsToPeriod(s), it.endField))
+    case x: IntegralType if it.startField == it.endField =>
```

Comment:
How does the error message look like when `t.startField != it.endField`. It would be nice if we say to users that according to the SQL standard the start and end fields should be the same.

Reply:
Oh, Maybe I made a mistake that we no need check `it.startField == it.endField`, because it has called the `canCast` and return false if `it.startField != it.endField` when go here. So I think no need add error message here. Do you think so?

Reply:
At now, if `it.startField != it.endField`. the exception is as follow like:
![image](https://user-images.githubusercontent.com/41178002/141419802-60483d96-80e1-48cb-a20d-ad6057dcf4f0.png)
![image](https://user-images.githubusercontent.com/41178002/141419842-0f02e271-827d-4468-9fec-e6bfe99b32bf.png)

Reply:
At now, if `it.startField != it.endField` the exception is as follow:
![image](https://user-images.githubusercontent.com/41178002/141420031-baf0bd4d-325d-415c-bb82-a185e4228712.png)
![image](https://user-images.githubusercontent.com/41178002/141420068-3099d80b-fc13-4e64-ae4c-4aa9bdeb4468.png)

Reply:
> So I think no need add error message here

+1